### PR TITLE
Allow custom configuration for apiserver host

### PR DESCRIPTION
**Pull request by ghost**
Currently your system uses the kubectl_util.DefaultClientConfig() to retrieve the IP for the api-server, which may not work outside GCE in customized setups.

This PR provides optional flags to use a non-default api-server. It uses the same flags and borrows code from the kubernetes/dashboard project:
https://github.com/kubernetes/dashboard/blob/master/src/app/backend/client/apiserverclient.go
https://github.com/kubernetes/dashboard/blob/master/src/app/backend/dashboard.go#L31

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1951)

### CODE REVIEW

**Comment by doug-clarifai** [comment]
I signed it!

**Comment by doug-clarifai** [comment]
I signed it!

File: ingress/controllers/nginx/apiserverclient.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"github.com/golang/glog"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+// Dashboard UI default values for client configs.
```
Comment (therc) [inline ingress/controllers/nginx/apiserverclient.go]: Reference to Dashboard

**Comment by bprashanth** [comment]
Is your goal to point at an apiserver that kubectl doesn't know about? because even outside a kubecluster, we should be looking up the default kubeconfig context, and pulling out the apiserver from there, at least the last time i checked

**Comment by aledbf** [comment]
@doug-clarifai this is not required. Please check this comment https://github.com/kubernetes/contrib/pull/1467#issue-168905136

**Comment by doug-clarifai** [comment]
@aledbf The comment you reference only allows a simple host name as an environment variable for the api server.

The goal of this change is to allow for more flexible configurations. kubernetes/dashboard added these flags to match kubectl and proxy: https://github.com/kubernetes/dashboard/pull/1256 https://github.com/kubernetes/dashboard/issues/905

**Pull request closed without merging**